src/plugins/FileService/components/Copyable.tsx

```diff
@@ -0,0 +1,39 @@
+import { useSnackbar } from 'notistack'
+import React from 'react'
+import { useI18N } from '../../../utils/i18n-next-ui'
+
+interface Props {
+    className?: string
+}
+
+export const CopyableCode: React.FC<Props> = ({ children, className }) => {
+    const { t } = useI18N()
+    const snackbar = useSnackbar()
+    const onSelect = (event: React.MouseEvent<Node>) => {
+        const selection = global.getSelection()
+        const range = global.document.createRange()
+        range.selectNode(event.currentTarget)
+        selection?.removeAllRanges()
+        selection?.addRange(range)
+    }
+    const onDeselect = () => {
+        global.getSelection()?.removeAllRanges()
+    }
+    const onCopy = (event: React.MouseEvent<Node>) => {
+        onSelect(event)
+        const success = global.document.execCommand('copy')
```

Comment:
You may want to use the modern API https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText

Reply:
OK

Reply:
(Note that a polyfill is needed for safari)

Reply:
your added the polyfill to master branch?